services/datamanager/datasync/upload_data_capture_file.go

```diff
@@ -33,20 +33,18 @@ func uploadDataCaptureFile(ctx context.Context, client v1.DataSyncServiceClient,
 	}
 
 	if md.GetType() == v1.DataType_DATA_TYPE_BINARY_SENSOR && md.GetMethodName() == datacapture.GetImages {
-		timeReq := sensorData[0].GetMetadata().GetTimeRequested()
-		timeRec := sensorData[0].GetMetadata().GetTimeReceived()
-		var getImgsRes pb.GetImagesResponse
-		mp := sensorData[0].GetStruct().AsMap()
-		if err := mapstructure.Decode(mp, &getImgsRes); err != nil {
+		var res pb.GetImagesResponse
+		if err := mapstructure.Decode(sensorData[0].GetStruct().AsMap(), &res); err != nil {
 			return err
 		}
+		captureTimestamp := res.GetResponseMetadata().GetCapturedAt()
```

Comment:
We should probably only use this if it's filled out, and otherwise fall back to timeReq and timeRec.

Reply:
Really great point, and hit this as soon as I tested with a fake camera. Updated.